### PR TITLE
Restore verbose kwarg to load_xdf

### LINKED ISSUE #41: Re-add verbose argument to `load_xdf`
The short of it is that I updated some Intheon code to no longer use 'verbose' in the call to `load_xdf`. Then @chkothe reverted that changed and pinned their use of pyxdf to an older version. So obviously this is important to them though I haven't had a conversation as to why.

Looking back, it seems that I'm the one that removed it [in this commit](https://github.com/sccn/xdf/commit/0428fe0aaa5f8565141342c98b3610e6f9d5ed55#diff-8dbbae27ec1074622bb151a61e0fb80e). The only thing related I see is [this comment](https://github.com/sccn/xdf/pull/6#issuecomment-263812175). So I guess my interpretation was "why use a 'verbose' argument when you should be setting the logger level separately?".

So if we re-add the verbose argument, does that mean we should use it to override the logger level? Or if verbose is set, do we skip the logger output and print to screen?

### PR DESCRIPTION
**Pull request by cboulay**
Fixes #41

### CODE REVIEW

**Comment by tstenner** [comment]
I'm not a fan of those `if verbose: log.debug()` calls. What about setting the default to `None` and if it's set changing the log level at the beginning and restoring it at the end?

**Comment by cbrnr** [comment]
+100 for doing it like @tstenner suggested. I'd implement the following behavior:

- If `verbose=None` (should be the default), we don't change the logging level and use whatever is currently set by the user.
- If `verbose=True`, we set the logging level to `INFO`.
- If `verbose=False`, we set the logging level to `WARNING`.
- In addition, string arguments are also possible, e.g. `verbose=DEBUG' sets the logging level to `DEBUG`. These string arguments are equivalent to passing `logging.DEBUG` etc.

MNE does it like that, see https://www.nmr.mgh.harvard.edu/mne/stable/generated/mne.set_log_level.html.

**Comment by cboulay** [comment]
I find it strange that INFO, which is technically above DEBUG, also includes DEBUG. Or am I getting that wrong?
Anyway, I force pushed the suggested changes. I think this is correct.

**Comment by cbrnr** [comment]
No, INFO does not include DEBUG. It's the other way round, DEBUG includes everything else.

File: pyxdf/pyxdf.py [hunk 1]
```diff
@@ -195,6 +200,8 @@ def load_xdf(filename,
         load the streams contained in a given XDF file
         >>> streams, fileheader = load_xdf('myrecording.xdf')
     """
+    if verbose is not None:
+        logger.setLevel(logging.INFO if verbose else logging.WARNING)
```
Comment (cbrnr) [inline pyxdf/pyxdf.py]: I think we should store the current log level and restore it at the end of the function. Maybe we could even write a `verbose` decorator for that purpose. This decorator would do all the context management and add a `verbose` argument to the function it wraps.

This is also my second comment - even if you do not want to implement that as a decorator, I think the `verbose` argument should be the last parameter of `load_xdf`.

Comment (tstenner) [reply pyxdf/pyxdf.py]: Or in `**kwargs`

Comment (cbrnr) [reply pyxdf/pyxdf.py]: Yep.

Comment (cbrnr) [reply pyxdf/pyxdf.py]: Example for decorator: https://www.nmr.mgh.harvard.edu/mne/dev/generated/mne.verbose.html

Comment (cboulay) [reply pyxdf/pyxdf.py]: Nope. I don't want to handle kwargs unless we're passing them on to something. There's no need for that here.

We also don't need a decorator for something that appears in exactly one function.

Please, less over-engineering.

**Comment by cboulay** [comment]
> No, INFO does not include DEBUG. It's the other way round, DEBUG includes everything else.

That's what I guessed from reading about logging levels in the docs though it was never explicit, but then I read something that stated the opposite explicitly. So, if DEBUG is the lowest-level, then why did you recommend that when verbose is provided we set it to INFO instead of DEBUG? Is it just a case of "that's common practice"?

> I think we should store the current log level and restore it at the end of the function.

Why? What's the use case? This logger is for this module only.

As far as I can tell, you're trying to prevent a situation where someone first calls load_xdf(verbose=False), then later call load_xdf(verbose=None) expecting the logger to use the root logging level, not the logging level that was set from the previous call.

If someone ever complains about that then by all means you can make a decorator. Decorate your heart out.

**Comment by cbrnr** [comment]
> That's what I guessed from reading about logging levels in the docs though it was never explicit, but then I read something that stated the opposite explicitly. So, if DEBUG is the lowest-level, then why did you recommend that when verbose is provided we set it to INFO instead of DEBUG? Is it just a case of "that's common practice"?

Yep.

> Why? What's the use case? This logger is for this module only.
>
> As far as I can tell, you're trying to prevent a situation where someone first calls load_xdf(verbose=False), then later call load_xdf(verbose=None) expecting the logger to use the root logging level, not the logging level that was set from the previous call.
>
> If someone ever complains about that then by all means you can make a decorator. Decorate your heart out.

Fair enough.

**Comment by cbrnr** [comment]
Feel free to change the behavior of `verbose=True` to setting the log level to DEBUG. I think that's what @chkothe had in mind in the first place. Other than that, +1 for merge.

**Comment by cboulay** [comment]
OK back to DEBUG. I'll run a couple more tests locally then merge.

**Pull request merged**